modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -317,3 +376,145 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return { hash, sighashType };
   }
 }
+
+/**
+ * Takes a partially signed transaction and removes the scripts and signatures.
+ *
+ * Inputs must be one of:
+ *  * P2PKH
+ *  * P2SH 2-of-3
+ *  * P2WSH 2-of-3
+ *  * P2WPKH
+ *  * P2SH -> P2WSH 2-of-3
+ *  * P2SH -> P2WPKH
+ *  * P2TR script path 2-of-2
+ *
+ * @param tx the partially signed transaction
+ *
+ * @return the removed scripts and signatures, ready to be added to a PSBT
+ */
+export function unsign(tx: UtxoTransaction<bigint>, prevOuts: TxOutput<bigint>[]): PsbtInputUpdate[] {
+  const ret = tx.ins.map((input, vin) => {
+    const prevOut = prevOuts[vin];
```

Comment:
I think we need to have some sort of check here that verifies that `prevouts.length` is equal to `tx.ins.length` otherwise this could be undefined, and we're accessing it on line 423 without any sort of undefined check.